alpaca/rest.go

```diff
@@ -1177,3 +1178,29 @@ func unmarshal(resp *http.Response, data interface{}) error {
 
 	return json.Unmarshal(body, data)
 }
+
+type localPlaceOrderRequest PlaceOrderRequest
+
+func (req PlaceOrderRequest) MarshalJSON() ([]byte, error) {
+	// marshal original POR
+	buf, err := json.Marshal(localPlaceOrderRequest(req))
+	if err != nil {
+		return nil, err
+	}
+
+	// unmarshal to allow updates
+	data := make(map[string]interface{})
+	if err = json.Unmarshal(buf, &data); err != nil {
+		return nil, err
+	}
+
+	// remove zero-value struct fields related to order size
+	if req.Notional == reflect.Zero(reflect.TypeOf(req.Notional)).Interface() {
```

Comment:
Is it okey to use `req.Notional.IsZero()` instead of reflection? Or am I missing something here?

Reply:
Also note that you can add the `omitempty` json tag to qty and notional to not include the zero value in the json, which I believe should make this custom marshaller not necessary? Or am I missing something?

Reply:
Unfortunately it doesn't work: the zero value of `decimal.Decimal` is marshalled to 0.

Reply:
oh right, that makes sense why this is required. in that case, I also agree `req.Notional.IsZero()` should work here instead of reflection I believe.